test/source/mock/fes/fes-constants.ts

```diff
@@ -141,7 +142,15 @@ export const processMessageFromUser3 = async (body: string, fesUrl: string) => {
     verificationPubs: [],
   });
   expect(decrypted.success).to.equal(true);
-  const decryptedMimeMsg = decrypted.content?.toUtfStr();
+  const decryptedMimeMsg = decrypted.content!.toUtfStr();
+  const regex = /cryptup-data="([^"]*)"/;
+  const match = decryptedMimeMsg.match(regex);
+  if (match && match.length > 1) {
+    const cryptupData = base64decode(match[1]);
+    const jsonObject = JSON.parse(cryptupData);
+    expect(jsonObject.recipient).to.not.include('flowcrypt.compatibility@gmail.com');
+    expect(cryptupData).to.contains('to@example.com');
+  }
```

Comment:
If there will be some bug and `cryptup-data` won't be found by regex, then this `if` part will be ignored, but test won't fail. That's why we should check values with `expect` instead of `if`. And let's also rename variables, instead of generic `regex` and `match`:
```ts
const cryptupDataRegex = /cryptup-data="([^"]*)"/;
const cryptUpDataMatch = decryptedMimeMsg.match(cryptupDataRegex);
expect(cryptUpDataMatch).to.not.be.undefined;
expect(cryptUpDataMatch.length).to.be.greaterThan(1);
const cryptupData = base64decode(match[1]);
...
expect(jsonObject.recipient).to.not.include('flowcrypt.compatibility@gmail.com');
expect(jsonObject.recipient).to.include('to@example.com');
```

Reply:
it's helpful to know - I will be considering that when testing data to ensure that the error will persist for variables that could potentially be `undefined` in case of an immediate bug appearing in code.